post-processor/upload/post-processor.go

```diff
@@ -0,0 +1,190 @@
+package upload
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/mitchellh/packer/common"
+	"github.com/mitchellh/packer/helper/config"
+	"github.com/mitchellh/packer/packer"
+	"github.com/mitchellh/packer/template/interpolate"
+	"golang.org/x/crypto/ssh"
+)
+
+type Config struct {
+	common.PackerConfig `mapstructure:",squash"`
+	Login               string `mapstructure:"login"`
+	Passw               string `mapstructure:"passw"`
+	KeyFile             string `mapstructure:"key"`
+	Endpoint            string `mapstructure:"endpoint"`
+	KeepInputArtifact   bool   `mapstructure:"keep_input_artifact"`
+	ctx                 interpolate.Context
+}
+
+type PostProcessor struct {
+	cfg Config
+}
+
+type Uploader interface {
+	Upload(packer.Artifact) error
+}
+
+type UploaderSSH struct {
+	Host    string
+	Port    string
+	Login   string
+	Passw   string
+	KeyFile string
+	Path    string
+	File    string
+	Timeout time.Duration
```

Comment:
Can we use an SSH communicator struct here so we can keep the same functionality, validation, and configuration scheme as other packer plugins?